icd/api/pipeline_compiler.cpp

```diff
@@ -1047,25 +1047,90 @@ void PipelineCompiler::GetPipelineCreationInfoNext(
+  {
+    if (pPipelineFeedback->feedbackValid)
+    {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+    }
+    if (pPipelineFeedback->hitApplicationCache)
+    {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+    }
+    pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->duration =
+        pPipelineFeedback->duration;
+    if (pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks !=
+            nullptr &&
+        pStageFeedback != nullptr)
+    {
+      // Compute pipeline does have VkGraphicsPipelineCreateInfo.
+      if (pCreateInfo == nullptr)
+      {
+        pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks[0]
+            .flags |= VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+        if (pStageFeedback[0].hitApplicationCache)
         {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
-                VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+          pPipelineCreationFeadbackCreateInfo
+              ->pPipelineStageCreationFeedbacks[0]
+              .flags |=
+              VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
         }
-        if (pPipelineFeedback->hitApplicationCache)
+        pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks[0]
+            .duration = pStageFeedback[0].duration;
+      }
+      else
+      {
+        VK_ASSERT(pCreateInfo->stageCount <= ShaderStage::ShaderStageGfxCount);
+        for (uint32_t i = 0; i < pCreateInfo->stageCount; ++i)
         {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          uint32_t feedbackStage = 0;
+          VK_ASSERT(pCreateInfo->pStages[i].sType ==
+                    VK_STRUCTURE_TYPE_PIPELINE_SHADER_STAGE_CREATE_INFO);
+          switch (pCreateInfo->pStages[i].stage)
+          {
+            case VK_SHADER_STAGE_VERTEX_BIT:
+              feedbackStage = ShaderStage::ShaderStageVertex;
+              break;
+            case VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT:
+              feedbackStage = ShaderStage::ShaderStageTessControl;
+              break;
+            case VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT:
+              feedbackStage = ShaderStage::ShaderStageTessEval;
+              break;
+            case VK_SHADER_STAGE_GEOMETRY_BIT:
+              feedbackStage = ShaderStage::ShaderStageGeometry;
+              break;
+            case VK_SHADER_STAGE_FRAGMENT_BIT:
+              feedbackStage = ShaderStage::ShaderStageFragment;
+              break;
+            default:
+              VK_ASSERT(false && "Invalid stage");
+              break;
+          }
+          pPipelineCreationFeadbackCreateInfo->
+              pPipelineStageCreationFeedbacks[i]
```

Comment:
Also here: can you create a reference to this array element?

Reply:
done